empress/__init__.py

```diff
@@ -0,0 +1,99 @@
+"""
+Wraps empress functionalities
+"""
+import matplotlib.pyplot as plt
+from typing import List, Iterable
+from abc import ABC, abstractmethod
+from empress.xscape.CostVector import CostVector
+
+class Drawable(ABC):
+    """
+    Implements a default draw method
+    """
+
+    @abstractmethod
+    def draw_on(self, axes: plt.Axes):
+        """
+        Draw self on matplotlib Axes
+        """
+        pass
+
+    def draw(self) -> plt.Figure:
+        """
+        Draw self as matplotlib Figure.
+        """
+        pass
+
+    def draw_to_file(self, fname):
+        """
+        Draw self and save it as image at path fname.
+        """
+        figure = self.draw()
+        figure.savefig(fname)
+
+class ReconInputWrapper:
+    def __init__(self, parasite_tree: dict, host_tree: dict, tip_mapping: dict):
+        self._parasite_tree = parasite_tree
+        self._host_tree = host_tree
+        self._tip_mapping = tip_mapping
+
+class ReconciliationWrapper(Drawable):
+    # TODO: Replace dict with Reconciliation type
+    # https://github.com/ssantichaivekin/eMPRess/issues/30
+    def __init__(self, reconciliation: dict):
+        self._reconciliation = reconciliation
+
+    def draw_on(self, axes: plt.Axes):
+        pass
+
+class ReconGraphWrapper(Drawable):
+    # TODO: Replace dict with ReconGraph type
+    # https://github.com/ssantichaivekin/eMPRess/issues/30
+    def __init__(self, recongraph: dict):
+        self._recongraph = recongraph
+
+    def find_median(self) -> ReconciliationWrapper:
+        """
+        Find and return one median of self
+        """
+        pass
+
+    def draw_on(self, axes: plt.Axes):
+        pass
+
+    def cluster(self, n) -> List['ReconGraphWrapper']:
+        """
+        Cluster self into n reconciliation graphs
+        """
+        pass
+
+class CostRegionWrapper(Drawable):
+    def __init__(self, info):
+        """
+        Will look into this more.
+        """
+        pass
+
+    def draw_on(self, axes: plt.Axes):
+        pass
+
+def read_tree(fname: str) -> ReconInputWrapper:
+    """
+    Read parasite tree, host tree, and tip mapping from fname.
+    Returns ReconInputWrapper which wraps the three info.
+    """
+    pass
+
+def compute_cost_region(recon_input: ReconInputWrapper) -> CostRegionWrapper:
```

Comment:
I don't know what the use case is for this one, but that's OK.  We can leave it here for now.

Reply:
This runs the reconcile algorithm in costcape behind the scene. It produces a list of cost vectors (and other information) which can be used for visualization.

Reply:
This is running costscape behind the scene. It produces a list of cost vectors and wraps it in the cost region wrapper.